### PR TITLE
SDIT-2237: ✨ Add in prisoner detail

### PR DESCRIPTION
**Pull request by petergphillips**

### CODE REVIEW

File: src/main/kotlin/uk/gov/justice/digital/hmpps/historicalprisonerapi/model/Prisoner.kt [hunk 1]
```diff
@@ -19,17 +19,17 @@ data class Prisoner(
   @GeneratedValue(strategy = GenerationType.IDENTITY)
   @Column(name = "PK_PRISONER", nullable = false)
   @JsonIgnore
-  val id: Int? = null,
+  val id: Int,
```
Comment (petergphillips) [inline src/main/kotlin/uk/gov/justice/digital/hmpps/historicalprisonerapi/model/Prisoner.kt:22]: unnecessary as not nullable

File: src/main/kotlin/uk/gov/justice/digital/hmpps/historicalprisonerapi/resource/PrisonerDetailResource.kt [hunk 2]
```diff
@@ -0,0 +1,43 @@
...
+        content = [Content(mediaType = "application/json", schema = Schema(implementation = ErrorResponse::class))],
+      ),
+      ApiResponse(
+        responseCode = "403",
+        description = "Forbidden to access this endpoint",
+        content = [Content(mediaType = "application/json", schema = Schema(implementation = ErrorResponse::class))],
+      ),
+    ],
+  )
+  fun getDetail(@PathVariable prisonNumber: String): String =
```
Comment (petergphillips) [inline src/main/kotlin/uk/gov/justice/digital/hmpps/historicalprisonerapi/resource/PrisonerDetailResource.kt:50]: I've mapped as `String` here but changed the `200` to include the actual dto.  Not sure it is worth hydrating to just  immediately serialise

File: src/main/kotlin/uk/gov/justice/digital/hmpps/historicalprisonerapi/model/PrisonerDetailDto.kt [hunk 3]
```diff
@@ -0,0 +1,140 @@
+package uk.gov.justice.digital.hmpps.historicalprisonerapi.model
+
+import com.fasterxml.jackson.annotation.JsonInclude
+import com.fasterxml.jackson.annotation.JsonInclude.Include.NON_NULL
+import java.time.LocalDate
+
+@JsonInclude(NON_NULL)
+data class PrisonerDetailDto(
```
Comment (petergphillips) [inline src/main/kotlin/uk/gov/justice/digital/hmpps/historicalprisonerapi/model/PrisonerDetailDto.kt:8]: none of this really needed mapping, but better for the API to return the types etc. as frontend needs it anyway

File: src/main/kotlin/uk/gov/justice/digital/hmpps/historicalprisonerapi/model/PrisonerDetail.kt [hunk 4]
```diff
@@ -0,0 +1,83 @@
...
+  @Size(max = 8)
+  @Column(name = "PK_PRISON_NUMBER", nullable = false, length = 8)
+  val pkPrisonNumber: String,
+
+  @NotNull
+  @Column(name = "PRISONER_DETAILS_ID", nullable = false)
+  val prisonerDetailsId: Int,
+
+  @Size(max = 512)
+  @Nationalized
```
Comment (mikehalmamoj) [inline src/main/kotlin/uk/gov/justice/digital/hmpps/historicalprisonerapi/model/PrisonerDetail.kt]: What's Nationalized required for? We don't have different character sets do we?

File: src/main/kotlin/uk/gov/justice/digital/hmpps/historicalprisonerapi/service/PrisonerDetailService.kt [hunk 5]
```diff
@@ -0,0 +1,33 @@
...
+        "hdcRecall": $hdcRecall,
+        "addresses": $addresses,
+        "adjudications": $adjudications,
+        "category": $category,
+        "courtHearings": $courtHearings,
+        "movements": $movements,
+        "offences": $offences,
+        "offencesInCustody": $offencesInCustody,
+        "sentencing": $sentencing
+      }"""
```
Comment (mikehalmamoj) [inline src/main/kotlin/uk/gov/justice/digital/hmpps/historicalprisonerapi/service/PrisonerDetailService.kt:31]: Seems a bit strange... why not use the ObjectMapper injected?

File: src/main/kotlin/uk/gov/justice/digital/hmpps/historicalprisonerapi/model/PrisonerDetail.kt (hunk 4 above)
Comment (petergphillips) [reply src/main/kotlin/uk/gov/justice/digital/hmpps/historicalprisonerapi/model/PrisonerDetail.kt]: thanks, that came through automatically by intellij and hadn't spotted it

File: src/main/kotlin/uk/gov/justice/digital/hmpps/historicalprisonerapi/resource/PrisonerDetailResource.kt (hunk 2 above)
Comment (mikehalmamoj) [reply src/main/kotlin/uk/gov/justice/digital/hmpps/historicalprisonerapi/resource/PrisonerDetailResource.kt:50]: Seems a bit odd but struggling to think of any negative consequences...

File: src/main/kotlin/uk/gov/justice/digital/hmpps/historicalprisonerapi/service/PrisonerDetailService.kt (hunk 5 above)
Comment (petergphillips) [reply src/main/kotlin/uk/gov/justice/digital/hmpps/historicalprisonerapi/service/PrisonerDetailService.kt:31]: I would just be using `ObjectMapper` to hydrate into `PrisonerDetailDto` only to then immediately afterwards convert back to `String` again.  Wasn't sure it was worth it?

**Review by mikehalmamoj — APPROVED** [review]

**Pull request merged**